### PR TITLE
CASSANDRASC-101 Upgrade Vert.x version to 4.5.3 

### PR DESCRIPTION
**Pull request by sarankk**
There was a memory leak identified in 4.4 branch in Vert.x  https://github.com/eclipse-vertx/vert.x/pull/5082. This leak is caused when connection pools are recreated for vert.x HttpClient. This was fixed in 4.5 branch, and a release was out 10 days ago. Sidecar is on 4.4.6 Vert.x version, hence we need to upgrade to 4.5.3 Vert.x version. As part of this we will also pick up dynamic update capability added for read and write thresholds set for servers https://github.com/eclipse-vertx/vert.x/pull/5088.

### CODE REVIEW

**Review by frankgh — APPROVED** [review]
+1 LGTM

File: src/main/java/org/apache/cassandra/sidecar/server/ServerVerticle.java [hunk 1]
```diff
@@ -114,6 +120,24 @@ Future<CompositeFuture> updateSSLOptions(SSLOptions options)
...
+     */
+    Future<Void> updateTrafficShapingOptions(TrafficShapingOptions options)
+    {
+        List<HttpServer> deployedServers = this.deployedServers;
+        if (deployedServers == null || deployedServers.isEmpty())
+        {
+            return Future.failedFuture("No servers are running");
+        }
+        return executorPools.service().executeBlocking(promise -> {
+            deployedServers.forEach(server -> server.updateTrafficShapingOptions(options));
```
Comment (frankgh) [inline src/main/java/org/apache/cassandra/sidecar/server/ServerVerticle.java]: this is not really blocking, everything is CPU bound

File: src/main/java/org/apache/cassandra/sidecar/server/Server.java [hunk 2]
```diff
@@ -189,6 +191,22 @@ public Future<CompositeFuture> updateSSLOptions(long timestamp)
         return Future.all(updateFutures);
     }
 
+    /**
+     * Updates the traffic shaping options for all servers in all the deployed verticle instances
+     *
+     * @param options update traffic shaping options
+     * @return a future to indicate the update was successfully completed
+     */
+    public Future<CompositeFuture> updateTrafficShapingOptions(TrafficShapingOptions options)
```
Comment (frankgh) [inline src/main/java/org/apache/cassandra/sidecar/server/Server.java]: can we add tests for this please?

**Review by frankgh — COMMENTED** [review]
I added some comments

**Review by frankgh — APPROVED** [review]
+1 looks good to me

File: src/main/java/org/apache/cassandra/sidecar/server/Server.java (hunk 2 above)
Comment (sarankk) [reply src/main/java/org/apache/cassandra/sidecar/server/Server.java]: sure, added 👍

**Comment by frankgh** [comment]
Closed via https://github.com/apache/cassandra-sidecar/commit/ca2af5e30cd58ffd3990069a5e46ef914a6a383a

**Pull request closed without merging**